internal/internal_schedule_client.go

```diff
@@ -637,6 +637,11 @@ func convertToPBScheduleAction(
 			return nil, err
 		}
 
+		userMetadata, err := buildUserMetadata(action.staticSummary, action.staticDetails, dataConverter)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
We need a similar change when the schedule is updated. The `UpdateSchedule` API on the server is structured to completely replace the `schedule.Action` with the value provided in the update API. So if the client wants to update a single field, it should echo back the whole schedule with the updated field. The client already seems to be doing that.
But in my testing I noticed that the usermetadata is not being copied over between these two calls - https://github.com/temporalio/sdk-go/blob/master/internal/internal_schedule_client.go#L276-L280. Looks like `ScheduleWorkflowAction` struct needs some updates?
cc @dnr

Reply:
This same code is used for update. The way update works in SDKs is it's a get-then-update which calls these same helpers to go from user model to proto model.